kcwitools/extract_optimal.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import print_function
+
+import numpy as np
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from kcwitools import optimal_extract_utility as r
+from linetools.spectra.xspectrum1d import XSpectrum1D
+
+
+def extract_optimal_1D(fluxcube, varcube,header,wavegrid,x1,x2,y1,y2,outfile=None,verbose=False,plot=False):
+	"""
+
+	Args:
+		fluxcube: 3D kcwi fluxcube
+		varcube: 3D KCWI variance cube
+		header:  header information
+		wavegrid:  1D kcwi wavelength grid
```

Comment:
If this can be trivially derived from the header, then no need to pass it in.

Reply:
Done